includes/Modules/Sign_In_With_Google/Web_Tag.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Sign_In_With_Google\Web_Tag
+ *
+ * @package   Google\Site_Kit\Modules\Sign_In_With_Google
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Sign_In_With_Google;
+
+use Google\Site_Kit\Core\Modules\Tags\Module_Web_Tag;
+use Google\Site_Kit\Core\Util\BC_Functions;
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+use Google\Site_Kit\Modules\Sign_In_With_Google\Authenticator;
+
+/**
+ * Class for Web tag.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Web_Tag extends Module_Web_Tag {
+
+	use Method_Proxy_Trait;
+
+	/**
+	 * Module settings.
+	 *
+	 * @since n.e.x.t
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * Context instance.
+	 *
+	 * @since n.e.x.t
+	 * @var \Google\Site_Kit\Context
+	 */
+	private $context;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string                   $tag_id      Tag ID.
+	 * @param string                   $module_slug Module slug.
+	 * @param Settings                 $settings    Module settings instance.
+	 * @param \Google\Site_Kit\Context $context     Context instance.
+	 */
+	public function __construct( $tag_id, $module_slug, Settings $settings, \Google\Site_Kit\Context $context ) {
+		parent::__construct( $tag_id, $module_slug );
+		$this->settings = $settings;
+		$this->context  = $context;
+	}
+
+	/**
+	 * Registers tag hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		// Render the Sign in with Google script that converts placeholder
+		// <div>s with Sign in with Google buttons.
+		add_action( 'wp_footer', $this->get_method_proxy( 'render_signinwithgoogle_scripts' ) );
+		// Output the Sign in with Google JS on the WordPress login page.
+		add_action( 'login_footer', $this->get_method_proxy( 'render_signinwithgoogle_scripts' ) );
+
+		$this->do_init_tag_action();
+	}
+
+	/**
+	 * Renders the Sign in with Google JS script tags, One Tap code, and
+	 * buttons.
+	 *
+	 * @since 1.139.0
+	 * @since 1.144.0 Renamed to `render_signinwithgoogle` and conditionally
+	 *                rendered the code to replace buttons.
+	 * @since n.e.x.t moved from main Sign_In_With_Google class to Web_Tag.
+	 */
+	protected function render_signinwithgoogle_scripts() {
```

Comment:
Is there a reason we don't use the `render` method here instead? It isn't hooked into anywhere by default but is intended for this purpose.